pkg/source/client.go

```diff
@@ -132,13 +132,19 @@ func MakeAdminClient(clientID string, kafkaAuthCfg *utils.KafkaAuthConfig, boots
 	saramaConf.Version = sarama.V2_0_0_0
 	saramaConf.ClientID = clientID
 
+	UpdateSaramaConfigWithKafkaAuthConfig(saramaConf, kafkaAuthCfg)
+
+	return sarama.NewClusterAdmin(bootstrapServers, saramaConf)
+}
+
+func UpdateSaramaConfigWithKafkaAuthConfig(saramaConf *sarama.Config, kafkaAuthCfg *utils.KafkaAuthConfig) {
 	if kafkaAuthCfg != nil {
 		// tls
 		if kafkaAuthCfg.TLS != nil {
 			saramaConf.Net.TLS.Enable = true
 			tlsConfig, err := NewTLSConfig(kafkaAuthCfg.TLS.Usercert, kafkaAuthCfg.TLS.Userkey, kafkaAuthCfg.TLS.Cacert)
 			if err != nil {
-				return nil, err
+				return
```

Comment:
any particular reason we're no longer passing the error back up?  It would probably be good to at least log it as an error for the user/admin

Reply:
it was before call insde a returning func. 

+1 on log - will update

Reply:
what if we changed the return type to an error, and returned nil in other cases, but then were able to `return fmt.Errorf("Error creating TLS config: %w, err")` and then the calling function can act accordingly?

Reply:
`fmt.Errorf` is not allowed - see the linter

Reply:
`fmt.Errorf` isn't allowed when the result isn't used, but if its used as the return value it's proper, no?

Reply:
updated - thanks!